benchmark/CMakeLists.txt

```
@@ -4,6 +4,13 @@ if (NOT CMAKE_BUILD_TYPE STREQUAL "Release")
         "will be affected")
 endif()
 
+if (GINKGO_BUILD_CUDA AND GINKGO_BUILD_HIP AND GINKGO_HIP_PLATFORM MATCHES "hcc")
+  message(FATAL_ERROR "Building the benchmarks for both HIP AMD and CUDA "
+          "at the same time is currently not supported. "
+	  "Disable the benchmark build using `-DGINKGO_BUILD_BENCHMARKS=OFF` "
+	  "or use `export HIP_PLATFORM=nvcc` in your build environment, instead.")
```

nit: Are you sure this comma is correct? If it was in the beginning of the sentence, a comma after it makes sense, but I have never seen it at the end of a sentence with a comma.
```suggestion
	  "or use `export HIP_PLATFORM=nvcc` in your build environment instead.")
```